data_model/src/events/data/events.rs

```diff
@@ -0,0 +1,301 @@
+//! This module contains data events
+
+use super::*;
+
+/// Trait for retrieving id from events
+pub trait IdTrait: Identifiable {
+    fn id(&self) -> &Self::Id;
+}
+
+mod asset {
+    //! This module contains `AssetEvent`, `AssetDefinitionEvent` and its impls
+
+    use super::*;
+
+    #[derive(Clone, PartialEq, Eq, Debug, Decode, Encode, Deserialize, Serialize, IntoSchema)]
+    #[non_exhaustive]
+    #[allow(missing_docs)]
+    pub enum AssetEvent {
+        Created(AssetId),
+        Deleted(AssetId),
+        Increased(AssetId),
+        Decreased(AssetId),
+        MetadataInserted(AssetId),
+        MetadataRemoved(AssetId),
```

Comment:
isn't metadata always related to `AssetDefinitionId`? I'm not totally certain, but I think this last two don't make sense for `AssetEvent`